### PR TITLE
INI inventory plugin: add documentation about variable types

### LINKED ISSUE #25784: Not all spellings of boolean "false" in inventory are interpreted as false by  Jinja
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME
inventory
template
jinja

##### ANSIBLE VERSION

```
ansible 2.3.0.0
  configured module search path = Default w/o overrides
  python version = 2.7.5 (default, Nov  6 2016, 00:28:07) [GCC 4.8.5 20150623 (Red Hat 4.8.5-11)]
```

##### CONFIGURATION
none special

##### OS / ENVIRONMENT
CentOS 7 on x86_64

##### SUMMARY
Boolean variables set in the inventory are not respected by Jinjia as one would expect.  According to http://docs.ansible.com/ansible/YAMLSyntax.html e.g. the boolean value false can be written "false", "False", "FALSE" or "no".  But only the exact spelling "False" is seen as false by Jinjia, all other spellings are seen as true value.

This is similar to and maybe related to (or the same problem as)  #25778.

##### STEPS TO REPRODUCE

```yaml
---
- hosts: all
  tasks:
  - shell: "echo \"The switch is {% if switch %}ON{% else %}OFF{% endif %} ({{ switch }})\""
    register: echo

  - debug:
      var: echo
```
set "switch" to various values in the inventory.

##### EXPECTED RESULTS

switch=False: The switch is OFF (False)
switch=FALSE: The switch is **OFF** (FALSE)
switch=false: The switch is **OFF** (false)
switch=no: The switch is **OFF** (no)
switch=0: The switch is OFF (0)
switch=true: The switch is ON (true)
switch=blah: _(not sure what to expect here, probably an error message)_

##### ACTUAL RESULTS
echo results:

switch=False: The switch is OFF (False)
switch=FALSE: The switch is **ON** (FALSE)
switch=false: The switch is **ON** (false)
switch=no: The switch is **ON** (no)
switch=0: The switch is OFF (0)
switch=true: The switch is ON (true)
switch=blah: The switch is ON (blah)

### PR DESCRIPTION
**Pull request by pilou-**
##### SUMMARY
Fixes #25784. INI inventory plugin [performs](https://github.com/ansible/ansible/blob/43468b825d19c5990c07d19f5f269d8352f084f8/lib/ansible/plugins/inventory/ini.py#L346) some undocumented processing on values of variables. This pull request aims to document and test the current behavior.

##### ISSUE TYPE
 - Test Pull Request
 - Docs Pull Request

##### COMPONENT NAME
Ini inventory plugin

##### ANSIBLE VERSION
```
ansible-playbook 2.4.0 (devel a38e727380) last updated 2017/06/16 15:40:05 (GMT +200)
```

### CODE REVIEW

File: docs/docsite/rst/intro_inventory.rst [hunk 1]
```diff
@@ -60,6 +60,9 @@ Suppose you have just static IPs and want to set up some aliases that live in yo
 In the above example, trying to ansible against the host alias "jumper" (which may not even be a real hostname) will contact 192.0.2.50 on port 5555.  Note that this is using a feature of the inventory file to define some special variables.  Generally speaking this is not the best
 way to define variables that describe your system policy, but we'll share suggestions on doing this later.  We're just getting started.
 
+.. note:: Values passed in using the ``key=value`` syntax are interpreted as Python literal structure (strings, numbers, tuples, lists, dicts,
```
Comment (bcoca) [inline docs/docsite/rst/intro_inventory.rst:63]: i would also note, that these are not guaranteed to survive templating as 'stringification happens', you cannot rely on the types ON DEFINITION, you should always use types on CONSUMPTION ( |list, |bool, |int, etc)

Comment (pilou-) [reply docs/docsite/rst/intro_inventory.rst:63]: When is 'stringification' performed ? Could you paste an example with unreliable type on definition ?

**Comment by bcoca** [comment]
any time it is templated
```
a=True
...
b: "{{a}}"
```

**Comment by pilou-** [comment]
Using the inventory and playbook below:
```
$ cat hosts
localhost a=(True,True)
```
```
$ cat play.yml
---
- hosts: all
  vars:
    b: '{{ a }}'
  tasks:
  - debug:
      msg: "{{ b }}"

  - debug:
      msg: "{{ b[0] is sameas true }}"

  - debug:
      msg: "{{ b | length }}"

  - debug:
      msg: "{{ [true, true] == b }}"

  - debug:
      msg: "{{ [true, true] is sameas b }}"

  - debug:
      msg: "{{ (true, true) == b }}"
```

The variable `b` is indeed an iterable but not a tuple:
```
$ ansible-playbook -i localhost, -i hosts play.yml 
PLAY [all] *

TASK [debug] *
ok: [localhost] => {
    "msg": [
        true, 
        true
    ]
}TASK [b[0] is sameas true] *
ok: [localhost] => {
    "failed": false, 
    "msg": true
}

TASK [b | length] *
ok: [localhost] => {
    "failed": false, 
    "msg": "2"
}

TASK [[true, true] == b] *
ok: [localhost] => {
    "failed": false, 
    "msg": true
}

TASK [[true, true] is sameas b] *
ok: [localhost] => {
    "failed": false, 
    "msg": false
}

TASK [(true, true) == b] *
ok: [localhost] => {
    "failed": false, 
    "msg": false
}
```

@bcoca would the sentence below be sufficient ?
```
Besides, the type of ``value`` could be modified when used in a template.
```

**Comment by bcoca** [comment]
`Do not rely on types set during definition, always make sure you specify type with a filter when needed when consuming the variable`

**Comment by pilou-** [comment]
Updated, thanks.

**Comment by pilou-** [comment]
ready_for_review

**Pull request merged**